src/openrct2/entity/Staff.cpp

```diff
@@ -1644,14 +1644,57 @@ bool Staff::UpdatePatrollingFindSweeping()
     return false;
 }
 
+int16_t Staff::CountNearbyPeeps() const
+{
+    // Iterate over tiles within a 3-tile radius (96 units)
+    constexpr auto kTileRadius = 3;
+    constexpr auto kLookupRadius = kCoordsXYStep * kTileRadius;
+
+    int16_t guestCount = 0;
+
+    for (int32_t tileX = x - kLookupRadius; tileX <= x + kLookupRadius; tileX += kCoordsXYStep)
+    {
+        for (int32_t tileY = y - kLookupRadius; tileY <= y + kLookupRadius; tileY += kCoordsXYStep)
+        {
+            for (auto* guest : EntityTileList<Guest>({ tileX, tileY }))
+            {
+                if (guest->x == kLocationNull)
+                    continue;
+
+                int16_t zDist = std::abs(z - guest->z);
+                if (zDist > 48)
```

Comment:
Maybe add a comment explaining how many tiles apart vertically this is?

Reply:
I think the z step size is half of the x and y step size. I think I based the loop on the `Staff::EntertainerUpdateNearbyPeeps` one, fwiw.